### PR TITLE
Jet 6440

### PR DESCRIPTION
**Pull request by wjtan99**
## Purpose

check in main.py that I use for training/val/test 

## Approach

How did you go about approaching the Jira acceptance criteria?
What noteworthy changes did you make that reviewers should be aware of?
Did you add new packages or change configurations?
Did you deviate from the original criteria of the Jira ticket?

## Questions

What feedback are you looking for from reviewers, general or do you have specific questions?

## TO-DOs

- A list of future items to do in order for the PR to go out.
- Links to follow-up tickets that should be done after this PR.

## Comments

Add anything else that doesn't fit into the categories above.

### CODE REVIEW

**Review by inder123 — COMMENTED** [review]
Let's try to do minimal changes in a third-party repository. Shall we close this PR and delete the branch? Thanks

**Review by zw-lvt — CHANGES_REQUESTED** [review]

File: main.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+
+if __name__ == "__main__":
+    ap = ap.ArgumentParser()
+    ap.add_argument("-d", "--dataset", default="dataset.yaml", help="path to the labeled dataset. default = dataset.yaml")
+    ap.add_argument("-b", "--batch_size", type=int, default=256, help="batch size for training. default = auto-batch")
+    ap.add_argument("-e", "--epochs", type=int, default=300, help="number of epochs to train for. default = 300")
+    #ap.add_argument("-f", "--freeze-layers", type=int, default=0, help="number of layers to freeze. default = 0")
+    ap.add_argument("-m", "--model", default="yolov8s.yaml", help="model to use for training. default = yolov8s")
+    ap.add_argument("-p", "--pretrained", default="yolov8s.pt", help="model to use for training. default = yolov8s")
```
[the remark this answers is not in the document]
Comment (wjtan99) [reply main.py:14]: Yes we need.  We have case where the model file is available but the pretrained checkpoint is not or not fully available.  One example is the yolov8s-P2.yaml. In that case we need both.

File: main.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    ap.add_argument("--conf",type=float, default = 0.1, help="confidence threshold for validation or prediction") 
+
+    args = ap.parse_args()
+    print('args = ', args) 
+
+    if args.disable_wandb:
+        os.environ['WANDB_MODE'] = 'disabled'
+
+    if args.job == 'validate' or args.job == 'test':
+        model = YOLO(args.pretrained)
```
[the remark this answers is not in the document]
Comment (wjtan99) [reply main.py:32]: There is some problem. I will fix it. 
We do need both -m and -p. See the first comment.

**Review by zw-lvt — APPROVED** [review]
Looks good to me.

**Pull request merged**